example/mapboxExampleCamera.js

```diff
@@ -0,0 +1,284 @@
+
+		const topHalfSurfaceDistance = ( Math.sin( fovAboveCenter ) * cameraToSeaLevelDistance ) / Math.sin(
+			utils.clamp(
+				Math.PI - groundAngle - fovAboveCenter,
+				0.01,
+				Math.PI - 0.01
+			)
+		);
+
+		// Calculate z distance of the farthest fragment that should be rendered.
+		furthestDistance = pitchAngle * topHalfSurfaceDistance + cameraToSeaLevelDistance;
+		// Add a bit extra to avoid precision problems when a fragment's distance is exactly `furthestDistance`
+		const horizonDistance = cameraToSeaLevelDistance * ( 1 / t._horizonShift );
+		farZ = Math.min( furthestDistance * 1.01, horizonDistance );
+
+		this.cameraTranslateZ = new THREE.Matrix4().makeTranslation(
+			0,
+			0,
+			this.cameraToCenterDistance
+		);
+		// someday @ansis set further near plane to fix precision for deckgl,so we should fix it to use mapbox-gl v1.3+ correctly
+		// https://github.com/mapbox/mapbox-gl-js/commit/5cf6e5f523611bea61dae155db19a7cb19eb825c#diff-5dddfe9d7b5b4413ee54284bc1f7966d
+		const nz = t.height / 50; // min near z as coded by @ansis
+		const nearZ = Math.max( nz * pitchAngle, nz ); // on changes in the pitch nz could be too low
+
+		const h = t.height;
+		const w = t.width;
+		this.camera.projectionMatrix = utils.makePerspectiveMatrix(
+			t._fov,
+			w / h,
+			nearZ,
+			farZ
+		);
+		// Unlike the Mapbox GL JS camera, separate camera translation and rotation out into its world matrix
+		// If this is applied directly to the projection matrix, it will work OK but break raycasting
+		const cameraWorldMatrix = this.calcCameraMatrix( t._pitch, t.angle );
+		// When terrain layers are included, height of 3D layers must be modified from t_camera.z * worldSize
+		if ( t.elevation )
+			cameraWorldMatrix.elements[ 14 ] = t._camera.position[ 2 ] * worldSize;
+		//this.camera.matrixWorld.elements is equivalent to t._camera._transform
+		this.camera.matrixWorld.copy( cameraWorldMatrix );
+
+		const zoomPow = t.scale * this.state.worldSizeRatio;
+		// Handle scaling and translation of objects in the map in the world's matrix transform, not the camera
+		const scale = new THREE.Matrix4();
+		const translateMap = new THREE.Matrix4();
+		const rotateMap = new THREE.Matrix4();
+
+		scale.makeScale( zoomPow, zoomPow, zoomPow );
+
+		const x = t.x || t.point.x;
+		const y = t.y || t.point.y;
+		translateMap.makeTranslation( - x, y, 0 );
+		rotateMap.makeRotationZ( Math.PI );
+
+		this.world.matrix = new THREE.Matrix4()
+			.premultiply( rotateMap )
+			.premultiply( this.state.translateCenter )
+			.premultiply( scale )
+			.premultiply( translateMap );
+
+	}
+
+	worldSize() {
+
+		const t = this.map.transform;
+		return t.tileSize * t.scale;
+
+	}
+
+	mercatorZfromAltitude( altitude, lat ) {
+
+		return altitude / this.circumferenceAtLatitude( lat );
+
+	}
+
+	circumferenceAtLatitude( latitude ) {
+
+		return (
+			MapConstants.EARTH_CIRCUMFERENCE *
+      Math.cos( ( latitude * Math.PI ) / 180 )
+		);
+
+	}
```

Comment:
This function looks like it's not used?

Reply:
Used in the mergetorZfromAltitude function